### ISSUE TITLE
gradle generateContractTests failes with jdk 17 with fromRequest().body('$.bar') in response.body

### ISSUE THREAD

**Issue by vgropp** [spring-cloud/spring-cloud-contract#1717]
We have several groovy contract tests in our projects. We are referencing the request json in some response validations `fromRequest().body('$.bar')`. Running gradle generateContractTests on a java 17 jdk failes with this Exception:
```
Exception in thread "main" wiremock.com.github.jknack.handlebars.HandlebarsException: inline@a585f17:534:64: java.lang.reflect.InaccessibleObjectException: Unable to make public boolean java.util.Collections$SingletonMap.isEmpty() accessible: module java.base does not "opens java.util" to unnamed module @3bc94c5e
    inline@a585f17:534:64
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
        at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
        at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
        at wiremock.com.github.jknack.handlebars.context.MemberValueResolver.cache(MemberValueResolver.java:82)
        at wiremock.com.github.jknack.handlebars.context.MemberValueResolver.resolve(MemberValueResolver.java:54)
        at wiremock.com.github.jknack.handlebars.Context$CompositeValueResolver.resolve(Context.java:199)
        at wiremock.com.github.jknack.handlebars.internal.path.PropertyPath.eval(PropertyPath.java:52)
        at wiremock.com.github.jknack.handlebars.Context$PathExpressionChain.next(Context.java:361)
        at wiremock.com.github.jknack.handlebars.Context$PathExpressionChain.eval(Context.java:381)
        at wiremock.com.github.jknack.handlebars.Context.get(Context.java:618)
        at wiremock.com.github.jknack.handlebars.Context.get(Context.java:672)
        at wiremock.com.github.jknack.handlebars.Context.get(Context.java:651)
        at wiremock.com.github.jknack.handlebars.Options.get(Options.java:631)
        at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.HandlebarsHelper.getRenderCache(HandlebarsHelper.java:80)
        at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.HandlebarsJsonPathHelper.getJsonDocument(HandlebarsJsonPathHelper.java:96)
        at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.HandlebarsJsonPathHelper.apply(HandlebarsJsonPathHelper.java:60)
        at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers$3.apply(WireMockHelpers.java:50)
        at org.springframework.cloud.contract.verifier.builder.handlebars.HandlebarsJsonPathHelper.apply(HandlebarsJsonPathHelper.java:59)
        at wiremock.com.github.jknack.handlebars.internal.Variable.value(Variable.java:167)
        at wiremock.com.github.jknack.handlebars.internal.Variable.merge(Variable.java:146)
        at wiremock.com.github.jknack.handlebars.internal.BaseTemplate.apply(BaseTemplate.java:125)
        at wiremock.com.github.jknack.handlebars.internal.TemplateList.merge(TemplateList.java:95)
        at wiremock.com.github.jknack.handlebars.internal.BaseTemplate.apply(BaseTemplate.java:125)
        at wiremock.com.github.jknack.handlebars.internal.BaseTemplate.apply(BaseTemplate.java:113)
        at wiremock.com.github.jknack.handlebars.internal.ForwardingTemplate.apply(ForwardingTemplate.java:72)
        at org.springframework.cloud.contract.verifier.template.HandlebarsTemplateProcessor.uncheckedApplyTemplate(HandlebarsTemplateProcessor.java:108)
        at org.springframework.cloud.contract.verifier.template.HandlebarsTemplateProcessor.templatedResponseBody(HandlebarsTemplateProcessor.java:103)
        at org.springframework.cloud.contract.verifier.template.HandlebarsTemplateProcessor.transform(HandlebarsTemplateProcessor.java:70)
        at org.springframework.cloud.contract.verifier.builder.TemplateUpdatingMethodPostProcessor.apply(TemplateUpdatingMethodPostProcessor.java:38)
        at org.springframework.cloud.contract.verifier.builder.TemplateUpdatingMethodPostProcessor.apply(TemplateUpdatingMethodPostProcessor.java:24)
        at org.springframework.cloud.contract.verifier.builder.SingleMethodBuilder.lambda$null$3(SingleMethodBuilder.java:201)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.LinkedList$LLSpliterator.forEachRemaining(LinkedList.java:1242)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at org.springframework.cloud.contract.verifier.builder.SingleMethodBuilder.lambda$build$4(SingleMethodBuilder.java:201)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at org.springframework.cloud.contract.verifier.builder.SingleMethodBuilder.build(SingleMethodBuilder.java:166)
        at org.springframework.cloud.contract.verifier.builder.ClassBodyBuilder.lambda$build$0(ClassBodyBuilder.java:73)
        at org.springframework.cloud.contract.verifier.builder.BlockBuilder.inBraces(BlockBuilder.java:166)
        at org.springframework.cloud.contract.verifier.builder.ClassBodyBuilder.build(ClassBodyBuilder.java:69)
        at org.springframework.cloud.contract.verifier.builder.GeneratedTestClassBuilder.build(GeneratedTestClassBuilder.java:156)
        at org.springframework.cloud.contract.verifier.builder.JavaTestGenerator.generatedTestClass(JavaTestGenerator.java:76)
        at org.springframework.cloud.contract.verifier.builder.JavaTestGenerator.classAsString(JavaTestGenerator.java:43)
        at org.springframework.cloud.contract.verifier.builder.JavaTestGenerator.buildClass(JavaTestGenerator.java:37)
        at org.springframework.cloud.contract.verifier.TestGenerator.processIncludedDirectory(TestGenerator.java:172)
        at org.springframework.cloud.contract.verifier.TestGenerator.lambda$processAll$1(TestGenerator.java:152)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
        at org.springframework.cloud.contract.verifier.TestGenerator.processAll(TestGenerator.java:151)
        at org.springframework.cloud.contract.verifier.TestGenerator.generateTestClasses(TestGenerator.java:141)
        at org.springframework.cloud.contract.verifier.TestGenerator.generate(TestGenerator.java:111)
        at org.springframework.cloud.contract.verifier.TestGeneratorApplication.main(TestGeneratorApplication.java:40)
```

Is there any workaround or fix so we can build on jdk 17, besides rewriting and using variables in both request and response body? springCloudVersion is 2020.0.4 with spring boot 2.5.5

**Sample**
```
package foobar

import org.springframework.cloud.contract.spec.Contract

REQUEST_URL = '/api/v0/foo'


[
    Contract.make {
        name("post_Happy")
        request {
            method 'POST'
            url REQUEST_URL
            body(
                    bar: 2985000,
            )
            headers {
                contentType(applicationJson())
            }
        }
        response {
            status 200
            body(
                    bar: fromRequest().body('$.bar'),
                    )
            headers {  contentType(applicationJson()) } }
    }
]

```

**Comment by marcingrzejszczak** [comment]
Jdk17 should be supported in 2021.0.0 rease train aka Jubilee. Please try using 3.1.0-M1.

**Comment by vgropp** [comment]
thanks for your feedback, i can confirm it will be fixed in 3.1.0-M1 of spring-cloud-starter-contract-verifier
